src/client/dfs/xattr.c

```diff
@@ -38,6 +38,8 @@ dfs_setxattr(dfs_t *dfs, dfs_obj_t *obj, const char *name, const void *value, da
 		return EINVAL;
 	if (name == NULL)
 		return EINVAL;
+	if (value == NULL && size != 0)
+		return EINVAL;
```

Comment:
same here. if the buffer is NULL, the size is totally ignored and should not be required to be init to 0.

Reply:
same here, for the case of (value == NULL && size != 0), below code at L67 "d_iov_set(&sg_iovs[0], (void *)value, size)" it will generate sgl with IOV of "iov.iov_buf = NULL, iov.iov_buf_len != 0" that cannot be allowed by client IO.